src/core/common/debug.hpp

```diff
@@ -47,7 +47,7 @@
 
 #define OT_ASSERT(cond) assert(cond)
 
-#elif OPENTHREAD_CONFIG_PLATFORM_ASSERT_MANAGEMENT
+#elif OPENTHREAD_CONFIG_PLATFORM_ASSERT_MANAGEMENT // defined(__APPLE__) || defined(__linux__)
```

Comment:
Same suggestion as before. This is less than 20 lines from `#if`

Reply:
The `#endif` is more than 20 lines from `#if`. I think the `if-elif-else-endif` block should be considered as a whole.

Reply:
The rational behind this is to help readers of code. Basically if we can see the `#if` condition itself and the comment line corresponding to `#if` in the same window/screen (i.e., they are less than 20 lines apart) we don't add the comment.

In this case, the `#if defined(__APPLE__) || defined(__linux__)` is just 6 lines above the newly added comment.
```
#if defined(__APPLE__) || defined(__linux__)

#include <assert.h>

#define OT_ASSERT(cond) assert(cond)

#elif OPENTHREAD_CONFIG_PLATFORM_ASSERT_MANAGEMENT // defined(__APPLE__) || defined(__linux__)
```

Reply:
Sounds reasonable. Updated.

Reply:
Thanks 👍